.githooks/pre-commit

```diff
@@ -1,10 +1,14 @@
 #!/bin/bash
-
-# pre-commit hook to update copyrights
+set -euo pipefail
 
 year=$(date +%Y)
 files=$(git diff --cached --name-only)
 
+# check code style
+git diff --cached --diff-filter=ACM -- '*.c' '*.h' | \
+	./scripts/checkpatch.pl --no-tree --ignore ELSE_AFTER_BRACE,SPDX_LICENSE_TAG -
```

Comment:
The script returns warnings and errors, so maybe block commit if there are any error? I just don't want to review pull requests with many style violations. (BTW #2157 also has many violations). @const-t I'd love to know your feedback on this.

Reply:
Yes, so that's why I set the bash options at the head of the script. It will fail out in case of any error.

https://github.com/tempesta-tech/tempesta/blob/80436fb92f38da6958cdcc5286c3c9729b82f422/.githooks/pre-commit#L2

Reply:
Looks good to me. I agree, we must not allow to commit patch that has warnings/error. In exceptional cases when `checkpatch.pl` gives false positive result committer can use `no-verify`, to bypass commit hook.